apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/service/TransportInteractionWebSocket.java

```diff
@@ -0,0 +1,381 @@
+                        this.tunaddr = transportStatusReply.tunaddr;
+                        this.tunnetmask = transportStatusReply.tunnetmask;                        
+                        log.debug("connected to: {} : {} via : {} / {}", remoteip, remoteport, tunaddr, tunnetmask);
+                        secureTransportStatus = SecureTransportStatus.CONNECTED;                        
+                    } else if (transportStatusReply.status.equals("INITIAL")) {
+                        // seems that we are dealing with service that have never been started.. Starting it up
+                        secureTransportStatus = SecureTransportStatus.NOT_LAUNCHED;
+                    }
+                    
+                } else if (type.equals("STARTREPLY")) {
+                    log.debug("StartReply, setting PENDING and waiting for connect event");
+                    secureTransportStatus = SecureTransportStatus.PENDING;
+                }                
+                
+            } else if (parent.has("event")) {
+                                
+                String eventSpec =  parent.get("event").asText();
+                if (eventSpec.equals("CONNECT") ) {
+                    // handling json here 
+                    TransportEventDescriptor transportEventDescriptor = processData(TransportEventDescriptor.class, message, false);
+                    this.remoteip= transportEventDescriptor.remoteip;
+                    this.remoteport = transportEventDescriptor.remoteport;
+                    this.tunaddr = transportEventDescriptor.tunaddr;
+                    this.tunnetmask = transportEventDescriptor.tunnetmask;                   
+                    log.debug("connected to: {} : {} via : {} / {}", remoteip, remoteport, tunaddr, tunnetmask);
+                    secureTransportStatus = SecureTransportStatus.CONNECTED;                    
+                } else if (eventSpec.equals("DISCONNECT")) {
+                    TransportEventDescriptor transportEventDescriptor = processData(TransportEventDescriptor.class, message, false);
+                    cleanupComParams();                    
+                    log.debug("disconnected from: {} : {} via : {} / {}", remoteip, remoteport, tunaddr, tunnetmask);
+                    secureTransportStatus = SecureTransportStatus.DISCONNECTED;                    
+                }
+            }
+        } catch (IOException ex) {
+            log.error("incoming message processing exception: {}", ex.toString());
+        }
+
+   
+    }
+    
+    /**
+     * Send a message.
+     *
+     * @param message
+     */
+    public void sendMessage(String message) {        
+        try {
+            Future<Void> fut;            
+            fut = session.getRemote().sendStringByFuture(message);
+            fut.get(2, TimeUnit.SECONDS);
+        } catch (InterruptedException ex) {
+            log.error ("sendMessage InterruptedException: {}", ex.getMessage().toString() );
```

Comment:
Looks like error handling is the same for all exceptions, combine it in one catch block